Content.Server/_Gabystation/CartridgeLoader/Cartridges/NanoBankCartridgeSystem.cs

```diff
@@ -0,0 +1,328 @@
+            return false;
+
+        pda = (container.Owner, pdaComp);
+        return true;
+    }
+
+    private void HandleNotification(EntityUid uid, string tittleLoc, string bodyLoc, int? amount = default, (string, object)? arg = default)
+    {
+        if (!TryPdaFromId(uid, out var pda))
+            return;
+
+        var args = new List<(string, object)>();
+
+        if (amount is not null)
+            args.Add(("amount", amount.Value));
+
+        if (arg is not null)
+            args.Add(arg.Value);
+
+        string body = Loc.GetString(bodyLoc, args.ToArray());
+
+        _cartridge.SendNotification(pda.Owner, Loc.GetString(tittleLoc), body);
+    }
+
+    private void HandleToggleMute(Entity<NanoBankCardComponent> card)
+    {
+        var foo = (card, card.Comp);
+        _nanoBank.SetNotificationsMuted(foo, !_nanoBank.GetNotificationsMuted(foo));
+        UpdateUIForCard(card);
+    }
+
+    private void HandleLogin(Entity<NanoBankCardComponent> card, int? id, int? pin)
+    {
+        if (!TryComp<EconomyManagerComponent>(card.Comp.Station, out var economy))
+            return;
+
+        if (id is null || pin is null)
+            return;
+
+        if (_economy.ValidateLogin(economy, id.Value, pin.Value))
+        {
+            card.Comp.LoggedIn = true;
+            card.Comp.AccountId = id.Value;
+            card.Comp.AccountPin = pin.Value;
+        }
+        else
+            card.Comp.LoggedIn = false;
+
+        Dirty(card);
+        UpdateUIForCard(card);
+    }
+
+    private void HandleTransfer(Entity<NanoBankCardComponent> card, int? targetAcc, int? amount)
+    {
+        if (targetAcc is null || amount is null ||
+            !TryComp<EconomyManagerComponent>(card.Comp.Station, out var economy) ||
+            !_economy.TransferBalance(economy, targetAcc.Value, card.Comp.AccountId, amount.Value))
+        {
+            HandleNotification(card.Owner, "economy-notification-transference-failed-title", "economy-notification-transference-failed-body", default);
+            return;
+        }
+
+        //HandleNotification(card.Owner, "economy-notification-transfer-title", "economy-notification-transfer-body", amount);
+        //UpdateUIForCard(card);
+    }
+
+    // Talvez isso não devese ser público. Mas preciso chamar em PdaSystem.
+    public void UpdateUIForCard(EntityUid cardUid)
+    {
+        // Os UpdateUI devem ser relativos ao accountId.
+        if (!TryPdaFromId(cardUid, out var pda)
+            || !_container.TryGetContainer(pda.Owner, SharedCartridgeLoaderSystem.InstalledContainerId, out var container))
+            return;
+
+        var nanoBankUid = container.ContainedEntities
+            .Where(HasComp<NanoBankCartridgeComponent>) // Pode acontecer do PDA ter mais de um nanobank instalado?
+            .First();
+
+        if (!TryComp<NanoBankCartridgeComponent>(nanoBankUid, out var nanoBankComp))
+            return;
```

Comment:
```suggestion
        var maybeNanoBankUid = container.ContainedEntities
            .Where(HasComp<NanoBankCartridgeComponent>) // Pode acontecer do PDA ter mais de um nanobank instalado?
            .FirstOrNull();

        if (maybeNanoBankUid is not { } nanoBankUid
            || !TryComp<NanoBankCartridgeComponent>(nanoBankUid, out var nanoBankComp))
            return;
```

acho que o que o cabrito escreveu faz sentido (dessa vez).